dotCMS/src/main/java/com/dotcms/browser/BrowserQuery.java

```diff
@@ -0,0 +1,221 @@
+            this.showFolders = browserQuery.showFolders;
+            this.showFiles = browserQuery.showFiles;
+            this.showPages = browserQuery.showPages;
+            this.sortByDesc = browserQuery.sortByDesc;
+            this.showLinks = browserQuery.showLinks;
+            this.showDotAssets = browserQuery.showDotAssets;
+            this.languageId = browserQuery.languageId;
+        }
+
+        public Builder withUser(@Nonnull User user) {
+            this.user = user;
+            return this;
+        }
+        
+        public Builder inHostOrFolder(@Nonnull Folder folder) {
+            this.hostFolderId = folder !=null ? folder.getInode() : null;
+            return this;
+        }
+        public Builder inHostOrFolder(@Nonnull Host host) {
+            this.hostFolderId = host.getIdentifier();
+            return this;
+        }
+        public Builder withHostId(@Nonnull String hostid) {
+            this.hostFolderId = hostid;
+            return this;
+        }
+        public Builder withHostOrFolderId(@Nonnull String hostFolderId) {
+            this.hostFolderId = hostFolderId;
+            return this;
+        }
+
+        public Builder withFilter(@Nonnull String filter) {
+            this.filter = filter;
+            return this;
+        }
+
+        public Builder showMimeTypes(@Nonnull List<String> mimeTypes) {
+            this.mimeTypes = mimeTypes;
+            return this;
+        }
+
+        public Builder showExtensions(@Nonnull List<String>  extensions) {
+            this.extensions = extensions;
+            return this;
+        }
+
+        public Builder sortBy(@Nonnull String sortBy) {
+            this.sortBy = sortBy;
+            return this;
+        }
+
+        public Builder offset(@Nonnull int offset) {
+            this.offset = offset;
+            return this;
+        }
+
+        public Builder maxResults(@Nonnull int maxResults) {
+            this.maxResults = maxResults;
+            return this;
+        }
+
+        public Builder showWorking(@Nonnull boolean showWorking) {
+            this.showWorking = showWorking;
+            return this;
+        }
+
+        public Builder showArchived(@Nonnull boolean showArchived) {
+            this.showArchived = showArchived;
+            return this;
+        }
+
+        public Builder showFolders(@Nonnull boolean showFolders) {
+            this.showFolders = showFolders;
+            return this;
+        }
+
+        public Builder showFiles(@Nonnull boolean showFiles) {
+            this.showFiles = showFiles;
+            return this;
+        }
+
+        public Builder sortByDesc(@Nonnull boolean sortByDesc) {
+            this.sortByDesc = sortByDesc;
+            return this;
+        }
+
+        public Builder showLinks(@Nonnull boolean showLinks) {
+            this.showLinks = showLinks;
+            return this;
+        }
+
+        public Builder showPages(@Nonnull boolean showPages) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'showPages' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)